apps/api/src/domain/games/memory-match/models/memory-match-round.validate-invariants.spec.ts

```diff
@@ -0,0 +1,280 @@
+                        cardBackImageId,
+                        cards: tooFewCards,
+                    });
+
+                    const result = roundWithTooFewCards.validateInvariants();
+
+                    expect(result).toHaveLength(1);
+
+                    assertErrorAsExpected(
+                        result[0],
+                        new MemoryRoundIsNotReadyForPublicationError(roundWithTooFewCards.id, [
+                            new InsufficientNumberOfCardsForPublicationError(
+                                roundWithTooFewCards.size,
+                                tooFewCards.length
+                            ),
+                        ])
+                    );
+                });
+            });
+
+            describe(`when one of the cards is missing its image`, () => {
+                it(`should return the expected error`, () => {
+                    const badSequenceNumber = 1;
+
+                    const emptyArray = Array(MAX_NUMBER_OF_CARDS);
+
+                    // TODO use `buildTestInstance`
+                    const invalidCards = emptyArray.fill(null).map((_, index) => {
+                        const sequenceNumber = index + 1;
+
+                        if (sequenceNumber === badSequenceNumber) {
+                            return new MemoryMatchCard({
+                                sequenceNumber,
+                                audioId: buildDummyUuid(index + 100),
+                                // imageId: buildDummyUuid(sequenceNumber + 200), // missing
+                                text: buildMultilingualTextWithSingleItem(
+                                    `text for card #${index}`
+                                ),
+                            });
+                        }
+
+                        const goodCard = buildCard(sequenceNumber);
+
+                        return goodCard;
+                    });
+
+                    const roundWithBadCard = new MemoryMatchRound({
+                        id: testRoundId,
+                        isPublished: true,
+                        cardBackImageId,
+                        cards: invalidCards,
+                    });
+
+                    const result = roundWithBadCard.validateInvariants();
+
+                    expect(result).toHaveLength(1);
+
+                    assertErrorAsExpected(
+                        result[0],
+                        new MemoryRoundIsNotReadyForPublicationError(roundWithBadCard.id, [
+                            new MissingImageForMemoryMatchCardError(badSequenceNumber),
+                        ])
+                    );
+                });
+            });
+
+            describe(`when one of the cards is missing its audio`, () => {
+                it(`should return the expected error`, () => {
+                    const badSequenceNumber = 1;
+
+                    const emptyArray = Array(MAX_NUMBER_OF_CARDS);
+
+                    // TODO use `buildTestInstance`
```

Comment:
Let's do this and any other cleanup to make this test more maintainable and readable.